test/Test/Toml/Parsing/Unit.hs

```diff
@@ -0,0 +1,146 @@
+{-# LANGUAGE FlexibleContexts #-}
+
+module Test.Toml.Parsing.Unit where
+
+import Test.Tasty.Hspec (Spec, context, describe, it)
+
+import Data.Text (singleton)
+import Test.Hspec.Megaparsec
+import Text.Megaparsec (parse)
+
+import Toml.Parser (arrayP, boolP, integerP)
+import Toml.Type (UValue (..))
+
+spec_Parser :: Spec
+spec_Parser = do
+  let parseX p given expected = parse p "" given `shouldParse` expected
+      failOn p given          = parse p "" `shouldFailOn` given
+
+  let parseArray  = parseX arrayP
+      parseBool   = parseX boolP
+      parseInt    = parseX integerP
+      arrayFailOn = failOn arrayP
+      boolFailOn  = failOn boolP
+
+  describe "arrayP" $ do
+    it "can parse arrays" $ do
+      parseArray "[]" []
+      parseArray "[1]" [UInt 1]
+      parseArray "[1, 2, 3]" [UInt 1, UInt 2, UInt 3]
+      parseArray "[1.2, 2.3, 3.4]" [UFloat 1.2, UFloat 2.3, UFloat 3.4]
+      parseArray "['x', 'y']" [UString (singleton 'x'), UString (singleton 'y')]
+      parseArray "[[1], [2]]" [UArray [UInt 1], UArray [UInt 2]]
+      -- TODO: Add a test for array of UDate after implementing issue #18
```

Comment:
Instead of writing this `TODO` you can use [`xit`](https://hackage.haskell.org/package/hspec-2.5.1/docs/Test-Hspec.html#v:xit) function `hspec` to mark this test as pending.

Reply:
@ChShersh  Using [`xit`](https://hackage.haskell.org/package/hspec-2.5.1/docs/Test-Hspec.html#v:xit) will make the test fail with the following message: `# PENDING: No reason given`, and I was trying to avoid failed tests.

Should I use `xit` anyway?

Reply:
@ghallak Well, it's not actually failing, it's pending. Using `xit` is better because after every test we will see those tests. And we can just enable them later very easily once we implement feature itself.